custom_components/ramses_cc/__init__.py

```diff
@@ -60,18 +61,22 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     )
     _LOGGER.debug("\r\n\nConfig = %s\r\n", config[DOMAIN])
 
+    broker = RamsesBroker(hass, config)
+    try:
+        await broker.start()
+    except TransportSerialError as exc:
+        _LOGGER.error("There is a problem with the serial port: %s", exc)
+        return False
+
+    hass.data.setdefault(DOMAIN, {})[BROKER] = broker
```

Comment:
It will init `hass.data[DOMAIN]` to `{}` if not already set. It would matter if multiple config entries where allowed as setup for one would not then wipe out data set by another. But in practice it's probably going to make more sense to only allow a single config entry otherwise there's the possibility that there will be a fight over the entities.

Reply:
It's functionally equivalent to:
```
if DOMAIN not in hass.data:
        hass.data[DOMAIN] = {}
hass.data[DOMAIN[BROKER] = broker
```

Reply:
OK, I have no idea how config flow works (this wouldn't be necessary with configuration.yaml) ) - so lets run with it.